airflow/configuration.py

```diff
@@ -418,55 +436,30 @@ def parameterized_config(template):
 
 log.info("Reading the config from %s", AIRFLOW_CONFIG)
 
-conf = AirflowConfigParser()
+conf = AirflowConfigParser(default_config=parameterized_config(DEFAULT_CONFIG))
 conf.read(AIRFLOW_CONFIG)
 
-
-def load_test_config():
-    """
-    Load the unit test configuration.
-
-    Note: this is not reversible.
-    """
-    conf.load_test_config()
-
 if conf.getboolean('core', 'unit_test_mode'):
-    load_test_config()
-
-
-def get(section, key, **kwargs):
-    return conf.get(section, key, **kwargs)
-
-
-def getboolean(section, key):
-    return conf.getboolean(section, key)
-
-
-def getfloat(section, key):
-    return conf.getfloat(section, key)
-
-
-def getint(section, key):
-    return conf.getint(section, key)
-
-
-def getsection(section):
-    return conf.getsection(section)
-
-
-def has_option(section, key):
-    return conf.has_option(section, key)
-
-
-def remove_option(section, option):
-    return conf.remove_option(section, option)
-
-
-def as_dict(display_source=False, display_sensitive=False):
-    return conf.as_dict(
-        display_source=display_source, display_sensitive=display_sensitive)
-as_dict.__doc__ = conf.as_dict.__doc__
-
+    conf.load_test_config()
 
-def set(section, option, value):  # noqa
-    return conf.set(section, option, value)
+# Historical convenience functions to access config entries
+
+load_test_config = conf.load_test_config
+get = conf.get
+getboolean = conf.getboolean
+getfloat = conf.getfloat
+getint = conf.getint
+getsection = conf.getsection
+has_option = conf.has_option
+remove_option = conf.remove_option
+as_dict = conf.as_dict
+set = conf.set # noqa
+
+for func in [load_test_config, get, getboolean, getfloat, getint, has_option,
+             remove_option, as_dict, set]:
+    _deprecated(
+        func,
+        "Accessing configuration method '{f.__name__}' directly from "
+        "the configuration module is deprecated. Please access the "
+        "configuration from the 'configuration.conf' object via "
+        "'conf.{f.__name__}'".format(f=func))
```

Comment:
Is this a new deprecation? (I think it is), and if so do we trigger this anywhere from within the Airflow code-base? If so probably best to make that a separate change.

If we don't currently trigger this warning from this repo then this change can stay.

Reply:
Looks like we would trigger this in a few places: `with timeout(configuration.getint('core', "DAGBAG_IMPORT_TIMEOUT")):` in airflow/models.py etc.

(I am -0.1 to this change, but don't mind it so long as we don't trigger our own deprecation warnings.)

Reply:
So it was suggested in the original PR that I based it on, I'm happy to remove the deprecation stuff if you want. See https://github.com/apache/incubator-airflow/pull/1862#issuecomment-260429818

Reply:
Ah cool, lets keep it in then, but try to fix up any references in the rest of the code to the deprecated versions.

Reply:
Ok, I made a separate commit for that so it is easier to review. I can squash them when reviewed.